### ISSUE TITLE
Cinnamon DE does NOT remember the position (left) of second monitor when I deactivate and reactivate it

### ISSUE THREAD

**Issue by tethragon** [linuxmint/cinnamon#9801]
```
 * Cinnamon version (cinnamon --version)
   - default PPA from Linux Mint 20
     (https://launchpad.net/~linuxmint-daily-build-team/+archive/ubuntu/daily-builds).
 * Linux mint 20
 * AMD Radeon 5600XT with default AMDGPU open source driver
 * 64 bit
  ```

**Issue**
I use a dual monitor setup (please see attached image)
![6a66e52dd13faca10952dd8ef960696f38c61709 png 560f43a25f7544b8136a7d2fa44b887e](https://user-images.githubusercontent.com/39238674/103234006-4095d900-4947-11eb-82e9-1810ab1de66f.png)

My second monitor is positioned left and my main monitor right. Sometimes I don't use my second monitor and I deactivate it clicking the "activate" toggle switch. The problem is than when I activate it again, the display manager (cinnamon-settings display) does NOT remember the previous position and it defaults the position of my second monitor again to the right. So, each time I deactivate and reactivate my second monitor, I need to position it again to the left. This is counter productive and I think it should be fixed.  

**Steps to reproduce**

1) open cinnamon-settings display
2) deactivate 2nd monitor
3) click apply
4) reactivate second monitor. Now the second monitor is positioned to the right, even though it was previously set to the left.

**Expected behaviour**

cinnamon-settings display should remember the last set monitors positions.

**Other information**

I managed to overcome this issue using a hand-made script

`xrandr --listactivemonitors | grep 'DisplayPort-0' && xrandr --output DisplayPort-0 --off || xrandr --output DisplayPort-0 --left-of HDMI-A-0 --auto
`

**Comment by ItzSwirlz** [comment]
I have a feeling that somewhere Cinnamon clears the monitor configuration and then applies. So here's what I want you to do.

Find ~/.cinnamon-monitors.xml. It may be in a subdirectory but find it.

First, ensure that the configuration is the way you want it. Then attach the file.

Now that the first file is uploaded, disable your monitor. This way I'll know what happens when you deactivate it.

Rename the cinnamon-monitors.xmlf ile to cinnamon-monitors-2.xml or some other way I can know this is after and then upload it.

**Comment by zaatas** [comment]
Note: I have the exact same problem.  However, my situation occurs only when I walk away for 20 minutes or so.  The Dell Precision (Mint 20) goes to sleep, and I return to wake it up.  Never occurs when I suspend the machine, only up wake up.
If there is any data I can provide let me know.

**Comment by tethragon** [comment]
> I have a feeling that somewhere Cinnamon clears the monitor configuration and then applies. So here's what I want you to do.
> 
> Find ~/.cinnamon-monitors.xml. It may be in a subdirectory but find it.
> 
> First, ensure that the configuration is the way you want it. Then attach the file.
> 
> Now that the first file is uploaded, disable your monitor. This way I'll know what happens when you deactivate it.
> 
> Rename the cinnamon-monitors.xmlf ile to cinnamon-monitors-2.xml or some other way I can know this is after and then upload it.

I found cinnamon-monitors.xml in this path: /home/george/.config/cinnamon-monitors.xml

I attach cinnamon-monitors.xml with both monitors active and the correct positioning
I also attach cinnamon-monitors-2.xml with my second monitor deactivated

I think the problem is here in the cinnamon-monitors-2.xml: 
`<x>0</x>`

Maybe it should be:
`<x>1920</x>`  

[cinnammon-monitors.zip](https://github.com/linuxmint/cinnamon/files/5750150/cinnammon-monitors.zip)

**Comment by ItzSwirlz** [comment]
So somewhere (not sure where) the information is being purged.

**Comment by tethragon** [comment]
Is there anything that can be done to fix the issue?

**Comment by anerdonamotorcycle** [comment]
For what it's worth, this exact behavior is also an issue on XFCE on Xubuntu as well, so I don't think it's any aspect of Cinnamon that's causing this.  I recently switched from Xubuntu to Mint as my "daily driver", and, like OP, I have a dual monitor setup with the left-hand monitor occasionally being used for other things.

**Comment by TxRxFx** [comment]
I'm also experiencing this on Centos.7.x with Cinnamon. 

Found the monitors.xml in ~/.config path and once I adjusted 2nd monitor to show on left (not right as it seems to default) I set the file to read-only. 

Now Display manager still opens up with that default on the right even though it's being forced to present on left so that tells me there is a lack of parity on the users previous settings and the display managers layout defaults.

Not likely to be Cinnamon, XFCE related I suspect. I'll reboot a few times and see if this helps me.

**Comment by tethragon** [comment]
Problem still exists in mint 20.3 cinnamon

**Comment by oerkel47** [comment]
Problem is still there and annoying the hell out of me.

**Comment by tethragon** [comment]
> Problem is still there and annoying the hell out of me.

I don't believe devs care to fix this :-( . I hope that there is a chance for this problem to be fixed, when mint fully transition to xWayland. Probably we have to w8 for 2-3 more years.

**Comment by oerkel47** [comment]
Yeah would be great if the transition to wayland would fix it.

**Comment by cptX** [comment]
Problem exists in LM22!!

**Comment by kenimou** [comment]
problem still exists. The weird thing is that if I plug in the hub to one perfect usb-c port, it forgets my configs almost every time. If I plug in a different usb-c port (which is less stable and often disconnects monitor), then it remembers the settings...

**Comment by tethragon** [comment]
A possible solution for this problem **for my case**, proposed to me by kimi AI is the following:

> Kimi:
> Great!
> Now let’s work-around the Cinnamon bug you linked (it still forgets the left-position when you disable / re-enable the second monitor) without waiting for an upstream fix.
> We’ll hook the re-enable event and instantly put the monitor back on the left side—every time you click the toggle in Settings → Display.
> 
> 🛠️ Auto-fix via udev + xrandr (works on-the-fly)
> 1. Create a tiny helper script
> 
> `sudo nano /usr/local/bin/cinnamon-monitor-fix.sh`
> 
> ```
> #!/bin/sh
> 
> # Re-set the layout if DP-1 (Dell) was just (re)connected
> 
> sleep 1        # give Cinnamon a moment to finish its own xrandr call
> export DISPLAY=:0
> export XAUTHORITY=/var/lib/lightdm/.Xauthority   # adjust if you use GDM/SDDM
> USER=$(who | grep '(:0)' | awk '{print $1}')
> XAUTHORITY=/home/$USER/.Xauthority DISPLAY=:0 \
> xrandr --output DisplayPort-1 --left-of DisplayPort-0 --auto
> ```
> Make it executable:
> 
> `sudo chmod +x /usr/local/bin/cinnamon-monitor-fix.sh`
> 
> 2. Add a udev rule that fires when the monitor re-appears
> 
> `sudo nano /etc/udev/rules.d/99-cinnamon-monitor-fix.rules`
> 
> `ACTION=="change", SUBSYSTEM=="drm", ENV{HOTPLUG}=="1", RUN+="/usr/local/bin/cinnamon-monitor-fix.sh"`
> Reload rules:
> 
> `sudo udevadm control --reload-rules
> `
> ✅ Result
> Whenever you re-enable the second monitor—via Cinnamon’s GUI toggle, hot-key, cable plug/unplug—the rule triggers the script and the monitor instantly jumps back to the left side, exactly as you had it before.
> 

I was wondering if this workaround could inspire somehow cinnamon devs to fix the problem.

**Comment by Marinell0** [comment]
Problem occurs to me and I'm on Ubuntu 24, so I don't think it's a cinnamon specific bug. Been trying to fix it for years with no success

**Comment by soumiknandi** [comment]
I use this script 

```
#!/bin/bash

# Display names
LAPTOP="eDP-1"
HDMI="HDMI-1"

# Get current monitor state
ACTIVE=$(xrandr --listactivemonitors)

# States:
# 1. Both enabled
# 2. Only laptop
# 3. Only HDMI

if echo "$ACTIVE" | grep -q "$LAPTOP" && echo "$ACTIVE" | grep -q "$HDMI"; then
    # BOTH -> LAPTOP ONLY
    xrandr \
        --output "$HDMI" --off \
        --output "$LAPTOP" --auto --primary

    notify-send "Display Mode" "Laptop display only"

elif echo "$ACTIVE" | grep -q "$LAPTOP"; then
    # LAPTOP ONLY -> HDMI ONLY
    xrandr \
        --output "$LAPTOP" --off \
        --output "$HDMI" --auto --primary

    notify-send "Display Mode" "HDMI display only"

elif echo "$ACTIVE" | grep -q "$HDMI"; then
    # HDMI ONLY -> BOTH
    xrandr \
        --output "$HDMI" --auto --primary --pos 0x0 \
        --output "$LAPTOP" --auto --right-of "$HDMI"

    notify-send "Display Mode" "Dual display mode"

else
    # FALLBACK
    xrandr \
        --output "$HDMI" --auto --primary --pos 0x0 \
        --output "$LAPTOP" --auto --right-of "$HDMI"

    notify-send "Display Mode" "Recovered dual display mode"
fi
```
